Core/GDCore/Events/CodeGeneration/EventsCodeGenerationContext.cpp

```diff
@@ -33,6 +35,22 @@ void EventsCodeGenerationContext::InheritsFrom(
             std::inserter(alreadyDeclaredObjectsLists,
                           alreadyDeclaredObjectsLists.begin()));
 
+  nearestAsyncParent = parent_.nearestAsyncParent;
+  asyncDepth = parent_.asyncDepth;
+  depthOfLastUse = parent_.depthOfLastUse;
+  customConditionDepth = parent_.customConditionDepth;
+  contextDepth = parent_.GetContextDepth() + 1;
+  if (parent_.maxDepthLevel) {
+    maxDepthLevel = parent_.maxDepthLevel;
+    *maxDepthLevel = std::max(*maxDepthLevel, contextDepth);
+  }
+}
+
+void EventsCodeGenerationContext::InheritsAsAsyncCallbackFrom(
+    const EventsCodeGenerationContext& parent_) {
+  parent = &parent_;
+  nearestAsyncParent = this;

```

Comment:
@arthuro555 this line seems weird to me. The nearest parent of an async callback is... itself?

Shouldn't this be either the parent, if async, or the nearest async parent of the parent (which can be null)?

Note that if we change this, we probably need to change the `InheritsFrom` too and the way we iterate on the `for` loops over EventsCodeGenerationContext (to start at `this`, rather than at the parent... I guess).

I'm asking because I think this is creating an issue where we don't find an object already declared in a parent

Reply:
I think this is normal, since it is the simplest way to have a parent tell all its children that it is the new nearest context as that pointer is inherited from a parent. This is used in two places: `ObjectsListNeeded`, in which we do want this context to be taken into account, as otherwise, the current async action would not know to pass down objects lists to the actions directly following it.

In `ShouldUseAsyncObjectsLists` though, it looks like that is an issue, yes. If an object has been used in the previous actions of the current context, it doesn't mean that we should suddenly start using async lists. 

What I propose is that instead of changing this line, which is correct and the most simple (as in performant) for most uses, I propose that instead, we change the for loop in `ShouldUseAsyncObjectsLists` to start at `nearestAsyncParent->parent->nearestAsyncParent` instead of `nearestAsyncParent`.

Reply:
Just made a commit for that, please tell me if I missed something I have not touched the code for about a month and might remember the full context around it incorrectly.